Firestore/core/test/unit/model/object_value_test.cc

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/model/object_value.h"
+
+#include "Firestore/core/src/model/value_util.h"
+#include "Firestore/core/src/remote/serializer.h"
+#include "Firestore/core/test/unit/testutil/testutil.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+const char kFooString[] = "foo";
+const char kBarString[] = "bar";
+
+namespace {
+
+using absl::nullopt;
+using testutil::DbId;
+using testutil::Field;
+using testutil::Map;
+using testutil::Value;
+
+class ObjectValueTest : public ::testing::Test {
```

Comment:
Note - can be compared to end result of https://github.com/firebase/firebase-android-sdk/pull/2494